addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -103,37 +124,33 @@ class MessageList extends Component {
      * component is patched. This is acceptable when <ThreadView/> has a
      * fixed height, which is the case for the moment. task-2358066
      */
-    async adjustFromComponentHints() {
-        if (!this.threadView) {
-            return;
-        }
-        if (!this.el) {
-            return;
-        }
-        for (const hint of this.threadView.componentHintList) {
+    adjustFromComponentHints() {
+        const { componentHintList, threadView } = this._lastRenderedValues();
+        for (const hint of componentHintList) {
             switch (hint.type) {
                 case 'change-of-thread-cache':
-                    this._adjustFromChangeOfThreadCache(hint);
-                    break;
                 case 'home-menu-hidden':
-                    this._adjustFromHomeMenuHidden(hint);
-                    break;
                 case 'home-menu-shown':
-                    this._adjustFromHomeMenuShown(hint);
-                    break;
-                case 'messages-loaded':
-                    this.threadView.markComponentHintProcessed(hint);
+                    // thread just became visible, the goal is to restore its
+                    // saved position if it exists or scroll to the end
+                    this._adjustScrollFromModel();
                     break;
                 case 'message-received':
-                    this._adjustFromMessageReceived(hint);
+                case 'messages-loaded':
+                case 'new-messages-loaded':
+                    // messages have been added at the end, either scroll to the
+                    // end or keep the current position
+                    this._adjustScrollForExtraMessageAtTheEnd();
                     break;
                 case 'more-messages-loaded':
-                    this._adjustFromMoreMessagesLoaded(hint);
-                    break;
-                case 'new-messages-loaded':
-                    this.threadView.markComponentHintProcessed(hint);
+                    // messages have been added at the start, keep the current
+                    // position
+                    this._adjustScrollForExtraMessageAtTheStart();
```

Comment:
`_adjustScrollForExtraMessage(s)AtTheX`

Reply:
What is the suggestion? Just add the "s" in both names?

Reply:
> Just add the "s" in both names?

Yes